Flank/build.gradle

```diff
@@ -22,6 +22,8 @@ tasks.withType(FindBugs) {
 
 dependencies {
     compile 'com.linkedin.dextestparser:parser:1.1.0'
+    compile 'com.samskivert:jmustache:1.13'
```

Comment:
we should specify `implementation` or `api` i think `compile` is deprecated.

Reply:
Well, I am not a Gradle expert but I think that since we are using the `java` plugin and not the `java-library` one, `compile` scope should be just fine.

Reply:
We don't want to be using the deprecated compile Gradle method. There's also a build performance win when using implementation.

Reply:
@bootstraponline AFAIK it's not deprecated. I can change the plugin to use the "java-library"  but, in addition to being out of scope for this PR, doesn't make much sense since Flank is not a library.

Reply:
I recommend reading about Gradle best practices to understand why this makes sense. Both Gradle and Google have posted blog articles on the topic. I think it's in scope for the PR. Technical debt isn't great.

Reply:
>I recommend reading about Gradle best practices to understand why this makes sense. Both Gradle and Google have posted blog articles on the topic. I think it's in scope for the PR. Technical debt isn't great.

Could you please elaborate? From my understanding using the `java-library` plugin is beneficial if you are building a library meant to included in other modules. That's why I don't see much benefit in make the change in this PR. Moreover, as I mentioned above, the `java` plugin is not deprecated so we are not adding more tech debt.

Reply:
```diff
diff --git a/Flank/build.gradle b/Flank/build.gradle
index 21febab..21e6840 100644
--- a/Flank/build.gradle
+++ b/Flank/build.gradle
@@ -21,10 +21,10 @@ tasks.withType(FindBugs) {
 }
 
 dependencies {
-    compile 'com.linkedin.dextestparser:parser:1.1.0'
-    compile 'com.samskivert:jmustache:1.13'
-    compileOnly 'com.google.code.findbugs:annotations:3.0.1'
-    testCompile group: 'junit', name: 'junit', version: '4.11'
+    implementation 'com.linkedin.dextestparser:parser:1.1.0'
+    implementation 'com.samskivert:jmustache:1.13'
+    compileOnly 'com.google.code.findbugs:annotations:3.0.1' // 'provided' doesn't exist in the java plugin
+    testImplementation group: 'junit', name: 'junit', version: '4.11'
 }
 
 jar {
diff --git a/gradle/gradle/wrapper/gradle-wrapper.properties b/gradle/gradle/wrapper/gradle-wrapper.properties
index 26b61d7..702c4b6 100644
--- a/gradle/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/gradle/wrapper/gradle-wrapper.properties
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.3.1-all.zip
```

See https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html for an explanation of the new dependency configurations. Also see the official Gradle java library docs here:
https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph

I have applied the recommended changes in the above diff.